staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -92,7 +93,8 @@ func (o *RecommendedOptions) ApplyTo(config *server.RecommendedConfig, scheme *r
 	if err := o.Authorization.ApplyTo(&config.Config.Authorization); err != nil {
 		return err
 	}
-	if err := o.Audit.ApplyTo(&config.Config); err != nil {
+	// need advice on how to plumb this in -- @sttts @deads2k
+	if err := o.Audit.ApplyTo(&config.Config, config.ClientConfig, config.SharedInformerFactory, utilevents.DefaultProcessInfo(), nil, nil); err != nil {
```

Comment:
@sttts @deads2k could you advise on how to plumb in these values? The definition is https://github.com/kubernetes/kubernetes/pull/67257/commits/69f41f95f0371c16986c52a8a93868b51a53b2e6#diff-515f037b063ce70c486bf4316e2f0789R582 should these be added to the generic server config?

Reply:
Doesn't look too bad actually. 

What is a bit strange is that `DynamicOptions.apply` returns a backend. Can we turn `func (o *AuditDynamicOptions) applyTo(c *server.Config, ...)` into `func (o *AuditDynamicOptions) createBackend(c server.Config, ...)` (note the removal of the pointer!) and plumb all the config in `AuditOptions.ApplyTo`. Side-effects plus return values always looks odd. Similar for the other options like `AuditLogOptions.applyTo`.
-